checkov/arm/runner.py

```diff
@@ -92,13 +92,36 @@ def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=R
 
                         skipped_checks = ContextParser.collect_skip_comments(resource)
 
-                        results = arm_registry.scan(arm_file, {resource_name: resource}, skipped_checks,
-                                                    runner_filter)
+                        results = arm_resource_registry.scan(arm_file, {resource_name: resource}, skipped_checks,
+                                                             runner_filter)
                         for check, check_result in results.items():
                             record = Record(check_id=check.id, check_name=check.name, check_result=check_result,
                                             code_block=entity_code_lines, file_path=arm_file,
                                             file_line_range=entity_lines_range,
                                             resource=resource_id, evaluations=variable_evaluations,
                                             check_class=check.__class__.__module__, file_abs_path=file_abs_path)
                             report.add_record(record=record)
+
+            if isinstance(definitions[arm_file], dict_node) and 'parameters' in definitions[arm_file].keys():
```

Comment:
@mikeurbanski1 Suggestion - merge the first part of the 2 `if`s - `if isinstance(definitions[arm_file], dict_node)` and then have 2 inner `if`s for parameters and resources

Reply:
Good call, thanks. Initially when I saw the comment referencing "nested ARM resources" I thought, "nope, not touching that!"

But it's much better this way (and still did not require touching that).